src/go/rpk/pkg/cli/cmd/redpanda/admin/brokers/brokers.go

```diff
@@ -62,14 +62,17 @@ func newListCommand(fs afero.Fs) *cobra.Command {
 			}
 			for _, b := range bs {
 				if b.IsAlive != nil {
-					headers = append(headers, "Is-Alive")
+					headers = append(headers, "Is-Alive", "Broker-Version")
 					orig := args
 					args = func(b *admin.Broker) []interface{} {
-						return append(orig(b), *b.IsAlive)
+						return append(orig(b), *b.IsAlive, b.Version)
 					}
 					break
 				}
 			}
+			if len(headers) == 3 {
+				headers = append(headers, "Broker-Version")
```

Comment:
Is broker version available in all versions?

Also, this is a bit buggy: if `IsAlive` is nil in all cases, then this block here adds the `Broker-Version` column but _does not_ add the argument.

If broker version is available in all versions, I'd change this to just adding the header always, and doing the orig/args-rewrite trick to always add the column.

Reply:
You're right. Given `Version` was introduced along with `IsAlive`, it's all-or-nothing things. That is, I didn't need to add that logic at all.